### PR TITLE
Update: Add column config drawer

### PR DESCRIPTION
**Pull request by jkusa**
## Description

Add column config drawer

## Proposed Changes

- Added navi-column-config component
- Added feature flag around report-view to integrate navi-column-config
- Cleaned up report-view integration test
- Add optional chaining
- Refactor visualization-toggle

## Screenshots

col

Screen Shot 2020-03-03 at 11 27 15 PM

Screen Shot 2020-03-03 at 10 43 24 PM

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by colemanProjects** [comment]
how will this look with the missing intervals warning? 
Screen Shot 2020-03-03 at 4 11 07 PM

File: packages/reports/addon/components/navi-column-config.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import Component from '@ember/component';
```
Comment (mattgraham1995) [inline packages/reports/addon/components/navi-column-config.js]: copyright

File: packages/reports/addon/templates/components/navi-column-config.hbs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (mattgraham1995) [inline packages/reports/addon/templates/components/navi-column-config.hbs]:
```suggestion
```

File: packages/reports/addon/templates/components/report-view.hbs [hunk 3]
```diff
@@ -1,45 +1,64 @@
...
+    {{else}}
+      <div class="report-view__visualization-container">
+        <ReportVisualization
+          class="report-view__visualization"
+          @report={{@report}}
+          @response={{@response}}
+          @container={{this}}
+          @annotationData={{this.annotationData}}
+          @isEditing={{and this.isEditingVisualization @hasRequestRun}}
+          @onUpdateReport={{pipe
```
Comment (mattgraham1995) [inline packages/reports/addon/templates/components/report-view.hbs]: Does this need to be `pipe`? Can it be changed to `queue`?

**Comment by jkusa** [comment]
@colemanProjects

> how will this look with the missing intervals warning?

It will be pushed over a little bit

> Maybe we could move the missing intervals warning above the visualization?

Potentially, however it might be a little distracting. We could also add margin on the right side so it can be "centered".

File: packages/reports/tests/integration/components/report-view-test.js [hunk 4]
```diff
@@ -64,157 +66,193 @@ module('Integration | Component | report view', function(hooks) {
...
+      .exists('Column config drawer displays "back" icon when open');
+
+    await click('.report-view__columns-button');
+    await animationsSettled();
+    assert
+      .dom('.navi-column-config__panel')
+      .doesNotExist('Column config drawer is closed after toggling the columns button');
+    assert
+      .dom('.report-view__columns-icon.fa-columns')
+      .exists('Column config drawer displays "column" icon when closedk');
```
Comment (mattgraham1995) [inline packages/reports/tests/integration/components/report-view-test.js]:
```suggestion
      .exists('Column config drawer displays "column" icon when closed');
```

File: packages/reports/tests/integration/components/report-view-test.js [hunk 5]
```diff
@@ -64,157 +66,193 @@ module('Integration | Component | report view', function(hooks) {
...
+
+    await click('.report-view__columns-button');
+    await animationsSettled();
+    assert.dom('.navi-column-config__panel').exists('Column config drawer is open after toggleing the columns button');
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-view-test.js]:
```suggestion
    assert.dom('.navi-column-config__panel').exists('Column config drawer is open after toggling the columns button');
```

File: packages/reports/app/styles/navi-reports/components/report-view.less [hunk 6]
```diff
@@ -98,4 +99,49 @@
...
+    flex-flow: column;
+    min-height: 0;
+    min-width: 0;
+  }
+
+  &__columns-button {
+    align-items: center;
+    background: white;
+    border: 1px solid @denali-gray-300;
+    border-left: 1px solid white;
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/app/styles/navi-reports/components/report-view.less]: ah, forgot to clean up 👍

File: packages/reports/addon/templates/components/report-view.hbs (hunk 3 above)
Comment (jkusa) [reply packages/reports/addon/templates/components/report-view.hbs]: Probably

File: packages/reports/addon/components/navi-column-config.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+@tagName('')
+@templateLayout(layout)
+class NaviColumnConfig extends Component {
+  /**
+   * @param drawerTransition - drawer transition
+   */
+  @action
+  *drawerTransition({ insertedSprites, removedSprites }) {
+    const offset = 500; // 2x the size of the drawer
+    const x = document.querySelector('.navi-column-config').getBoundingClientRect().left - offset;
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/navi-column-config.js]: Consider that, but we don't have element-modifiers in yet.

File: packages/reports/app/styles/navi-reports/components/report-view.less [hunk 8]
```diff
@@ -98,4 +99,49 @@
...
+    border-left-width: 0;
+    border-radius: 0px 45% 45% 0px;
+    box-shadow: rgba(60, 64, 67, 0.3) 2px 1px 4px -1px;
+    display: flex;
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/app/styles/navi-reports/components/report-view.less]: Good catch

File: packages/reports/addon/components/report-view.js [hunk 9]
```diff
@@ -69,6 +71,11 @@ class ReportView extends Component {
       .join(' ');
   }
 
+  /**
+   * @property {Boolean} isEditingVisualization - Display column config or not
```
Comment (balajimanoharan) [inline packages/reports/addon/components/report-view.js]: update?

File: packages/reports/addon/templates/components/report-view.hbs [hunk 10]
```diff
@@ -1,45 +1,64 @@
...
-      @onUpdateReport={{pipe
-        (route-action "onUpdateReport")
-        (route-action "validate" @report)
-        (route-action "runReport" @report)
-      }}
-    />
-    <MissingIntervalsWarning
-      @response={{this.response}}
-      @onDetailsToggle={{this.resizeVisualization}}
+  <div class="report-view__visualization-header">
```
Comment (balajimanoharan) [inline packages/reports/addon/templates/components/report-view.hbs]: div -> section

File: packages/reports/addon/components/report-view.js (hunk 9 above)
Comment (jkusa) [reply packages/reports/addon/components/report-view.js]:
```suggestion
   * @property {Boolean} isColumnDrawerOpen - Display column config or not
```

File: packages/reports/addon/templates/components/report-view.hbs (hunk 10 above)
Comment (jkusa) [reply packages/reports/addon/templates/components/report-view.hbs]: Not sure `<section>` is the right tag here. 

> section denotes a semantic section of your content to help construct a document outline

File: packages/reports/app/styles/navi-reports/components/visualization-toggle.less [hunk 11]
```diff
@@ -1,42 +1,31 @@
...
+    border-radius: 4px;
+    color: @navi-blue-300;
     cursor: pointer;
-
-    &:hover {
-      background-color: @navi-lightest-gray;
-    }
+    font-size: 16px;
+    padding: 2px 12px;
+    transition: 1s ease;
```
Comment (kevinhinterlong) [inline packages/reports/app/styles/navi-reports/components/visualization-toggle.less]:
```suggestion
    transition: 0.5s ease;
```

**Pull request merged**